shared/settings.cpp

```diff
@@ -0,0 +1,61 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <QDebug>
+#include <QFile>
+#include "settings.h"
+
+Settings::Settings(): _deviceType(DeviceType::RM1) {
+    readDeviceType();
+}
+
+void Settings::readDeviceType() {
+    QFile file("/sys/devices/soc0/machine");
+    if(!file.open(QIODevice::ReadOnly | QIODevice::Text)){
+        qDebug() << "Couldn't open " << file.fileName();
+        return;
```

Comment:
```suggestion
    if(!file.exists || !file.open(QIODevice::ReadOnly | QIODevice::Text)){
        qDebug() << "Couldn't open " << file.fileName();
        _deviceType = DeviceType::Unknown;
        return;
```

Reply:
The file.exists is missing arguments and I doubt whether it's needed.

Reply:
Or braces at least.

Reply:
Whoops, thought I added the parenthesis.

Reply:
What it is for? If the file doesn't exists file.open will return false anyway...

Reply:
Being explicit and not relying on potentially undefined behaviour. The QFile.open documentation doesn't specify that it will fail if the file doesn't exist.